Refactor validation

MattElcock opened the pull request:
> BEFORE MAKING YOUR PR... There must be no linting errors and all tests must pass

### What is the context of this PR?

> Describe what you have changed as part of this task and why; include a link to the Trello card/GitHub issue, related pull requests, etc as required.
>
> E.g. _"Currently, if you add a new question to Author the entire app crashed. This is because..."_

### How to review 

> Add to the list below as appropriate, including screenshots when necessary

* Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
    * ensure it can be opened in Author;
    * then, ensure it can be viewed in Runner by pressing the **view survey** button
* Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Trello card for this task into the next stage of the process

tjamesmac commented:
Hey @MattElcock, was just checking this out in case I got to review it and it won't let build the API locally with Docker. It's saying it can't find `prettier`. Is that happening for you?

tjamesmac commented:
All looks really nice so far though, nice one!

MattElcock commented:
> Hey @MattElcock, was just checking this out in case I got to review it and it won't let build the API locally with Docker. It's saying it can't find `prettier`. Is that happening for you?

Hm... Not for me. I'll have a look at it once I'm done, 'though, to make sure!

> All looks really nice so far though, nice one!

Thanks!! :D I'll do a demo on it soon for feedback.

MattElcock commented:
PR is ready to be pulled apart by code monkeys. I've updated the description with as much detail as I can, but ask away if there are any questions. I've also squashed the commits as there was a loooot.

MattElcock commented:
I've had a look at the Codacy errors where it's saying the brackets in case statements aren't needed. If I take them out, the linter which lives in the code complains. Here's the rule that not having them violates: https://eslint.org/docs/rules/no-case-declarations.

If anyone objects, I'm going to ignore Codacy for those errors.

MattElcock commented:
When creating Skip Logic, the red dot is on the wrong tab (From Martyn; placed here so I don't forget)

tjamesmac commented:
I hit comment way too fast. Basically a bunch of constants and alternatives but not essentials.

Other than that awesome job. I haven't approved it yet because I'm not 100% that I can replicate the issues Martyn is having. Can definitely have a look again though.

MattElcock commented:
> I hit comment way too fast. Basically a bunch of constants and alternatives but not essentials.
> 
> Other than that awesome job. I haven't approved it yet because I'm not 100% that I can replicate the issues Martyn is having. Can definitely have a look again though.

Cheers mate!!! I'll sort those issues now. I've fixed what Martyn has found; if he hasn't explained already, to replicate his issue:

1. Create a new questionnaire with a single section and question
2. Add a number answer and fill in the label with whatever you want
3. Go to the logic tab and create a new routing rule
4. Add the number answer to that routing rule
5. Go back to the design tab and hit refresh
6. The editor pane will say 'Something went wrong'

Or, it did say that! I've fixed the issue. So, it should work now.

jasehumphr reviewed (approved):
Nice One Matt. Finally got there
